internal/proxy/directors.go

```diff
@@ -246,18 +241,26 @@ func (h DirectorSetHeadersByIP) Director(request *http.Request) error {
 
 	ip := net.ParseIP(host)
 
-	for _, ipHeaders := range h {
-		if !ipHeaders.IPNet.Contains(ip) {
-			continue
+	for _, headerName := range h.allHeaders {
+		_, ok := request.Header[headerName]
+		if ok && h.shouldRemoveHeaderByIP(headerName) {
```

Comment:
No need check h.shouldRemoveHeaderByIP, because we have range loop by headers to remove.